salt/metalk8s/orchestrate/bootstrap/init.sls

```diff
@@ -0,0 +1,141 @@
+        kwarg={
+            'key': 'metalk8s:control_plane_ip',
+        },
+    )[pillar.bootstrap_id]['ret']
+%}
+
+{%- if 'metalk8s' in pillar
+        and 'nodes' in pillar.metalk8s
+        and pillar.bootstrap_id in pillar.metalk8s.nodes
+        and pillar.metalk8s.nodes[pillar.bootstrap_id].version is not none %}
+{% set version = pillar.metalk8s.nodes[pillar.bootstrap_id].version %}
+{% else %}
+{% set _, version = saltenv.split('-', 1) %}
+{% endif %}
+
+{%- set pillar_data = {
+        'bootstrap_id': pillar.bootstrap_id,
+        'registry_ip': bootstrap_control_plane_ip,
+        'metalk8s': {
+            'nodes': {
+                pillar.bootstrap_id: {
+                    'roles': ['bootstrap', 'master', 'etcd','ca'],
+                    'version': version,
+                },
+            },
+            'endpoints': {
+                'salt-master': {
+                    'ip': bootstrap_control_plane_ip,
+                    'port': 4507,
+                },
+                'registry': {
+                    'ip': bootstrap_control_plane_ip,
+                    'port': 5000,
+                }
+            },
+        },
+    }
+%}
+
+Sync all custom types:
+  salt.runner:
+  - name: saltutil.sync_all
+  - saltenv: {{ saltenv }}
+
+Sync roster extmods:
+  salt.runner:
+  - name: saltutil.sync_roster
+  - saltenv: {{ saltenv }}
+
+Sync auth extmods:
+  salt.runner:
+  - name: metalk8s_saltutil.sync_auth
+  - saltenv: {{ saltenv }}
+
+Sync bootstrap minion:
+  salt.function:
+  - name: saltutil.sync_all
+  - tgt: {{ pillar.bootstrap_id }}
+
+Update mine from bootstrap minion:
+  salt.function:
+  - name: mine.update
+  - tgt: {{ pillar.bootstrap_id }}
+
+Deploy CA role on bootstrap minion:
+  salt.state:
+  - tgt: {{ pillar.bootstrap_id }}
+  - sls:
+    - metalk8s.roles.ca
+  - saltenv: metalk8s-{{ version }}
+  - require:
+    - salt: Sync bootstrap minion
+    - salt: Update mine from bootstrap minion
+
+Bring bootstrap minion to highstate:
+  salt.state:
+  - tgt: {{ pillar.bootstrap_id }}
+  - highstate: true
+  - pillar: {{ pillar_data | tojson }}
+  - require:
+    - salt: Sync bootstrap minion
+    - salt: Update mine from bootstrap minion
+    - salt: Deploy CA role on bootstrap minion
+
+Wait for API server to be available:
+  http.wait_for_successful_query:
+  - name: https://{{ pillar.metalk8s.api_server.host }}:6443/healthz
+  - match: 'ok'
+  - status: 200
+  - verify_ssl: false
+  - require:
+    - salt: Bring bootstrap minion to highstate
+
+Update bootstrap Node object:
+  salt.runner:
+  - name: state.orchestrate
+  - mods:
+    - metalk8s.kubernetes.mark-control-plane.deployed
+  - saltenv: {{ saltenv }}
+  - pillar: {{ pillar_data | tojson }}
+  - require:
+    - http: Wait for API server to be available
+
+Update pillar on bootstrap minion after highstate:
+  salt.function:
+  - name: saltutil.refresh_pillar
+  - tgt: {{ pillar.bootstrap_id }}
+  - require:
+    - salt: Configure bootstrap Node object
```

Comment:
This require is a left over I think ?

Reply:
No. The Pillar `top` is derived from the node/minion roles, and these are derived from annotations on the `Node` object, which needs to be configured correctly before we can continue.

Reply:
Yes indeed just when I commented the step "salt.runner Configure bootstrap Node object" just above was not here :stuck_out_tongue: